### ISSUE TITLE
BUG: argparse options not properly listed in report

### ISSUE THREAD

**Issue by tylerjereddy** [darshan-hpc/darshan#495]
See comment here: https://github.com/darshan-hpc/darshan/pull/485/files#r714438280

`python -m darshan summary --help` not showing some of our custom options:

```
usage: darshan <command>

PyDarshan CLI Utilities

optional arguments:
  -h, --help  show this help message and exit
  --debug
  --version

```

**Comment by tylerjereddy** [comment]
Wouldn't be surprised if placing the `argparse` machinery right after `__name__ ..` block cleared things up a bit instead of current abstraction.

**Comment by tylerjereddy** [comment]
Something like this has always seemed more natural to me: https://stackoverflow.com/a/27957417/2942522

**Comment by jakobluettgau** [comment]
Oh good catch! Thats a simple fix fortunetly :)
Just a missing "add_help=False" to the preparser.

I opened a pull request to at least fix the --help being swallowed, as there is an easy fix for that: https://github.com/darshan-hpc/darshan/pull/496


However, moving or rethinking some of the argparse machinery to __name__ sounds fair to me.

What I had in mind is having a "subcommand" style CLI interface (darshan <subcommand> ..., while also allowing python -m darshan.cli.subcommand ...) that is quite popular with say git and other projects to have a simple to remember entry point. 

(That being said, adding a additional __name__ == '__main__' blocks to places like the plots or other analysis still seems useful to me and almost makes them a little self-documenting).


On the technical side:
Currently, we used argparses subcommand functionally, which being the facility that argparse offers for such cases seemed reasonable use. This does require some convention how to register the add_arguments with the subparser, hence the setup_parser function, which was the least intrusive while staying reasonable close to how argparse is used in simpler setups. 

There is also one drawback to the current approach, as we add commands, more and more subparsers need to be initialized even though they might never be needed. We could potentially replace this with a simpler mechanic that just passes argparse.REMAINDER to the subcommand. (That does not completely fix the issue of putting the add_argument in a dedicated easy to invoke place).
